themes/default/content/blog/serverless-architecture-design/index.md

```diff
@@ -0,0 +1,116 @@
+    - system-architecture
+
+---
+
+Serverless architectures have become popular because they make it easier for software developers to focus on application development without the additional operation overhead. With Pulumi, you can deploy and manage your serverless infrastructure and this article shows you different architectural design considerations for your serverless application.
+
+Before we go into the specifics of serverless architecture design, let's go through the basics of serverless - this is important for the rest of the article.
+
+## What is Serverless Architecture
+
+Serverless architecture is a cloud computing model that simplifies the management of infrastructure. This allows software developers to focus on writing code without worrying about server provisioning, scaling or maintenance. The idea behind it is functions as a service (FaaS), in which the environment dynamically scales in response to demand and code is executed in response to events. In the serverless architecture, developers focus on writing and deploying code in the form of serverless functions, which are executed in response to specific events or triggers.
+
+Different cloud providers have their different serverless offerings. Azure has Azure Functions, AWS has Lambda, and Google Cloud has Google Cloud Functions. These are serverless platforms that allow developers to build event-driven functions that respond to various triggers. In addition to these major players, other cloud providers like Oracle and IBM also offer their own serverless solutions, creating a diverse ecosystem of options for developers to choose from. These serverless platforms enable software developers to build and deploy applications that respond to triggers, such as HTTP requests, database changes, and more, while abstracting away the complexities of infrastructure management.
+
+Using serverless eliminates the responsibilities of operational management, which streamlines the development process. This means you don't have to set up complex security mechanisms or firewalls. In addition, serverless computing abstracts away all operational complexities, saving you from the complexities of operating system patching and guaranteeing a simplified and trouble-free development process.
+
+Key characteristics of serverless architecture include:
+
+- Event-Driven: Serverless functions are triggered by events in the system. These events could be HTTP requests, messages in a queue, or changes to a data store.
+
+- Auto-Scaling: With serverless, resources automatically scale up or down to match the workload so you don't use infrastructure beyond what you actually need.
+
+- Pay-as-You-Go: You are charged only for the actual compute resources you use during code execution.
+
+The serverless architecture model is in line with current tech trends such as cloud-native development and DevOps, by allowing developers to focus on code development and pushing faster releases.
+
+## Serverless Function Design
```

Comment:
Is there code somewhere? like does it work and can you follow along with it and deploy it?